@xen-orchestra/xapi/host.mjs

```diff
@@ -31,7 +33,36 @@ class Host {
    *
    * @param {string} ref - Opaque reference of the host
    */
-  async smartReboot($defer, ref) {
+  async smartReboot($defer, ref, bypassBlockedSuspend = false) {
+    let currentVmRef
+    try {
+      currentVmRef = await this.call('VM.get_by_uuid', await getCurrentVmUuid())
+    } catch (error) {}
+
+    const residentVmRefs = await this.getField('host', ref, 'resident_VMs')
+    const vmsWithSuspendBlocked = await asyncMap(residentVmRefs, async ref => {
+      // No use of the spread operator to preserve non enumerable properties
+      const vm = await this.getRecord('VM', ref)
+      vm.ref = ref
+      return vm
+    }).filter(
+      vm =>
+        vm.ref !== currentVmRef &&
+        !vm.is_control_domain &&
+        vm.power_state !== 'Halted' &&
+        vm.power_state !== 'Suspended' &&
+        vm.blocked_operations.suspend !== undefined
+    )
+
+    if (!bypassBlockedSuspend && vmsWithSuspendBlocked.length > 0) {
+      throw incorrectState({ actual: vmsWithSuspendBlocked.map(vm => vm.uuid), expected: [], object: 'suspendBlocked' })
+    }
+
+    await asyncEach(vmsWithSuspendBlocked, vm => {
+      $defer(() => vm.update_blocked_operations('suspend', vm.blocked_operations.suspend ?? null))
```

Comment:
If the current VM is on the host being suspended, it will not be suspended because that would prevent the host from being rebooted, we count on the fact that the XOA is auto power on.

But that would make this defer not being run after the reboot, therefore the suspended VMs will not be resumed nor their `suspend` blocked operations reset properly.

We should discuss with @olivierlambert if this behavior is acceptable.